openmetadata-service/src/main/java/org/openmetadata/service/security/saml/SamlAssertionConsumerServlet.java

```diff
@@ -98,6 +106,8 @@ private void handleResponse(HttpServletRequest req, HttpServletResponse resp) th
                     ServiceTokenType.OM_USER);
       } catch (Exception e) {
         LOG.error("[SAML ACS] User not found: " + username);
+        // Create the user
+        user = UserUtil.addOrUpdateUser(UserUtil.user(username, email.split("@")[1], username));
```

Comment:
can  we centralize the email.split in addOrUpdate instead of adding it here. We need to gracefully handle this . We are expecting it to be email all the time